### PR TITLE
Implemented missing logic for noMoveStart when using panTo function

### PR DESCRIPTION
**Pull request by Chivano**
Fixed the open issue #5396 where Map.panTo didnt respect the option noMoveStart as documented in the API documentation

### CODE REVIEW

File: src/map/Map.js [hunk 1]
```diff
@@ -203,7 +203,7 @@ export var Map = Evented.extend({
 		}
 
 		// animation didn't start, just reset the map view
-		this._resetView(center, zoom);
+		this._resetView(center, zoom, options.pan);
```
Comment (johnd0e) [inline src/map/Map.js:206]:
```suggestion
		this._resetView(center, zoom, options.pan && options.pan.noMoveStart);
```

File: src/map/Map.js [hunk 2]
```diff
@@ -1181,7 +1181,8 @@ export var Map = Evented.extend({
 	// private methods that modify map state
 
 	// @section Map state change events
-	_resetView: function (center, zoom) {
+	_resetView: function (center, zoom, options) {
+		options = options || {};
```
Comment (johnd0e) [inline src/map/Map.js:1185]:
```suggestion
	_resetView: function (center, zoom, noMoveStart) {
```

File: src/map/Map.js [hunk 3]
```diff
@@ -1192,7 +1193,7 @@ export var Map = Evented.extend({
 
 		var zoomChanged = this._zoom !== zoom;
 		this
-			._moveStart(zoomChanged, false)
+			._moveStart(zoomChanged, options.noMoveStart || false)
```
Comment (johnd0e) [inline src/map/Map.js:1196]:
```suggestion
			._moveStart(zoomChanged, noMoveStart)
```

Comment (aniss) [reply src/map/Map.js:1196]: Is there any one want to take this fix?

**Comment by Falke-Design** [comment]
@Chivano can you please add the suggested changes

**Review by Falke-Design — APPROVED** [review]

**Pull request merged**